tar_scm.py

```diff
@@ -72,6 +72,17 @@ def fetch_upstream_git(url, clone_dir, revision, cwd, kwargs):
     """Fetch sources via git."""
     safe_run(['git', 'clone', url, clone_dir], cwd=cwd,
              interactive=sys.stdout.isatty())
+    if revision:
+        # check if the reference already exists.
+        try:
+            safe_run(['git', 'rev-parse', '--verify', '--quiet', revision],
```

Comment:
I can change it to ``subprocess.call`` resulting in some code duplication from ``safe_run`` but I just would like to let you know that you are using it in the same way in ``update_cache_hg``. I tried not to duplicate code and not to break the existing style.

Therefore I would like to know if this is really what you want me to do?